spinel/codec.py

```diff
@@ -119,7 +119,10 @@ def parse_e(cls, payload):
 
     @classmethod
     def parse_U(cls, payload):
-        payload = payload.decode('utf-8')
+        try:
+            payload = payload.decode('utf-8')
+        except UnicodeDecodeError as e:
+            payload = payload[:e.start].decode('utf-8')
```

Comment:
It seems not a good idea to fix it such way. 

some questions:
- What does the format `UD` mean?
- Why do the spinel message contains undecodable bytes after the log string?
- Can spinel get where the string ends without having a decoding error?

Reply:
```
 *   |   "U"    | DATATYPE_UTF8        | Zero-terminated UTF8-encoded    |
 *   |          |                      | string.                         |
 ```
If `U` is zero-terminated UTF8 string, should we search for '\0' first?

Reply:
- In `UD`, `U` means utf-8 while `D` means data. 
- The data part of a log line contains metadata. See https://github.com/openthread/openthread/blob/main/src/lib/spinel/spinel.h#L3519. I don't have much idea why spinel protocol was designed in this way. Apparently it's not easy to figure out the exact boundary of these two parts.

Reply:
> * Apparently it's not easy to figure out the exact boundary of these two parts.

It's not difficult. We can search for the 0 byte, which indicates the end of the UTF8 string.